exp/controllers/awsmachinepool_controller.go

```diff
@@ -140,6 +139,21 @@ func (r *AWSMachinePoolReconciler) Reconcile(ctx context.Context, req ctrl.Reque
 		return ctrl.Result{}, err
 	}
 
+	// Create the launch template scope
+	launchTemplateScope, err := scope.NewLaunchTemplateScope(scope.LaunchTemplateScopeParams{
```

Comment:
Do we need a new scope at this level? We have the `MachinePoolScope` is the scope of the reconciliation of the machine pool, so personally i don't think we need it here. But there is nothing stopping us creating the scope further down.

Reply:
I would potentially change `ec2Svc.ReconcileLaunchTemplate` so that it accepts the machine pool scope like we do for the [NodegroupService](https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/main/pkg/cloud/services/eks/service.go#L72).

Reply:
Good suggestion! I refactored the code so we don't create a new scope here and in the AWSMMP controller.